### DISCUSSION TITLE
Show a placeholder avatar image?

### DISCUSSION THREAD

**Discussion by YashDVardhan** [flyerhq/flutter_chat_ui#306] [Q&A] [answered]
Hi, I was wondering if there was a way to show placeholder avatar images for slower connections. I also wanted to know how Images are being cached so they don't have to be downloaded from the backend repeatedly (I saw usages of NetworkImage for avatar images but was wondering how normal image messages work with caching). Thanks for any help!

**Comment by demchenkoalex** [comment]
Hey! You have `avatarBuilder` available if you're not satisfied with the default one, so you can use any cached images and placeholders there :)

**Comment by demchenkoalex** [comment, marked as answer]
should be cached by default when added in v2 https://github.com/flyerhq/flutter_chat_ui/issues/646
